File: Sorting/insertion.py
def insertion_sort(a):
    n = len(a)
    for i in range(n):
        j = i-1
        temp = a[i]

        while j>=0 and temp < a[j]:
            a[j+1] = a[j]
            j-=1
        
        a[j+1] = temp
        
a = [5, 27, 10, 3, 0, 1, -15, 99, 8, 2]
print(a)
insertion_sort(a)
print(a)

File: Sorting/insertion.c
#include<stdio.h>

void swap(int*a, int*b){
    int temp = *a;
    *a = *b;
    *b = temp;
}

void InsertionSort(int a[], int n){
    for(int i=0; i<n; i++){
        int j = i-1;
        int element = a[i];

        while(j>=0 && element < a[j]){
            a[j+1] = a[j];
            j--;
        }

        a[j+1] = element;
    }
}

void printArray(int a[], int size){
    for (int i=0; i<size; i++){
        printf("%d\t", a[i]);
    }
    printf("\n");
}

int main(){
    int a[] = {4, 3, 7, 1, 5, 8, 2, 6, 10, 9};

    printArray(a, sizeof(a)/sizeof(a[0]));
    InsertionSort(a, sizeof(a)/sizeof(a[0]));
    printArray(a, sizeof(a)/sizeof(a[0]));

    return 0;
}